G2/G3 extrusion add

ochkariky opened the pull request:
Аdded the possibility of extrusion when processing circles (G2-G3).
Tested with Сura arc welder plugin on delta printer.

ochkariky committed:
G2/G3 extrusion add

Tested with Сura arc welder plugin.

wolfmanjm commented:
This functionality was added last year https://github.com/Smoothieware/Smoothieware/commit/2a68f14c3811053f243be7e33cb9c82ca55ba421

arthurwolf commented:
Thanks for the effort though. I'm a bit confused how you coded this into Smoothie when it's already in there, did you maybe not test if it worked and just presumed it didn't?

ochkariky commented:
> Thanks for the effort though. I'm a bit confused how you coded this into Smoothie when it's already in there, did you maybe not test if it worked and just presumed it didn't?

Now I'm surprised! I used the firmware like December 2020. A few months ago I tried to make the Cura start code to clean the nozzle in a circle. I failed - the effector made a circle but there was no extrusion. After looking at some old thread that G2/G3 is not supported, I calmed down.
But a week ago, I remembered it and afterwards added the code by analogy with linear movement - and everything seemed to work. Maybe I did something wrong last time and the code I added now is a dummy?
I don't know much about Smoothie's source code, which I added - of course a crutch)
PS maybe I took the wrong firmware binary, but I seem to be checking..

wolfmanjm commented:
FWIW this would not actually of added any motion to the Extruder during a G2/G3, all it added was a check on volumetric throttling (which actually maybe a good idea to add).

ochkariky commented:
> FWIW this would not actually of added any motion to the Extruder during a G2/G3, all it added was a check on volumetric throttling (which actually maybe a good idea to add).

Funny, now I see)))) Why didn't it work the first time... Maybe i used the wrong call format. excuse me, i'm new to this business)
I can safely delete this Fork, is it useless?
_PS off topic - work on Linear/Pressure Advance is not expected?_

arthurwolf commented:
On Wed, Jun 2, 2021 at 5:00 PM ochkariky ***@***.***> wrote:

> FWIW this would not actually of added any motion to the Extruder during a
> G2/G3, all it added was a check on volumetric throttling (which actually
> maybe a good idea to add).
> Funny, now I see)))) Why didn't it work the first time... Maybe i used the
> wrong call format.
>
> I can safely delete this Fork, is it useless?
>

bits are cheap, and this PR makes reference to the code, don't delete
anything, just leave it alone.

> *PS off topic - work on Linear/Pressure Advance is not expected?*
>

We tend to think that's the slicer's job, and we haven't seen any
firmware-side implementations around that were clean/good enough to
integrate into Smoothie, but we're open to ideas.

—
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Smoothieware/Smoothieware/pull/1509#issuecomment-853101989>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA2SFILIOOHMUUADGEQFZDTQZBP3ANCNFSM455RM6JA>
> .
>


-- 

勇気とユーモア
